spec/castle/middleware/configuration_spec.rb

```diff
@@ -4,12 +4,35 @@
 
 describe Castle::Middleware::Configuration do
   subject(:instance) { described_class.new }
+
   describe '#reset!' do
     before { instance.api_secret = 'secret' }
 
     it { expect { instance.reset! }.to change(instance, :api_secret).to nil }
   end
 
+  describe '#error_handler' do
+    let(:api) { double }
+
+    before do
+      instance.error_handler = handler
+      ::Castle::Middleware.configuration = instance
+      allow(::Castle::API).to receive(:new).and_return(api)
+      allow(api).to receive(:request).and_raise(::Castle::Error)
+    end
+
+    context 'when handler configured' do
+      let(:handler) { spy }
```

Comment:
Move this one block up for better readability?

Reply:
The idea was that this is specific to the context described, ie. `handler` is defined